File: test/webdir/test/model/Customer.js
Ext.define('Test.model.Customer', {
    extend: 'Ext.data.Model',
    fields: [
        {name: 'customer_id',  type: 'string'},
        
        {name: 'sales_id',  type: 'string'},
        
        {name: 'customer_name',  type: 'string'},
        {name: 'customer_type',  type: 'string'},
        {name: 'payment_type',  type: 'string'},
        {name: 'customer_contact',  type: 'string'},
        {name: 'customer_phone',  type: 'string'},
        {name: 'customer_fax',  type: 'string'},
        {name: 'customer_email',  type: 'string'},
        {name: 'customer_address',  type: 'string'},
        {name: 'createdDate',  type: 'date'},
        {name: 'lastModifiedDate',  type: 'date'}
    ],
    

    proxy: {
        type: 'rest',
        api: {
            read: 'CT/004/customerList.json',
            update: 'CT/update.json',
            create: 'CT/create.json',
            destroy: 'CT/delete.json'
        },
        simpleSortMode:true,
        reader: {
            type: 'json',
            root: 'stats',
            successProperty: 'success'
        }
    }
});

File: test/src/com/erp/china/demo/model/Customer.java
package com.erp.china.demo.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;

@Entity
@Table(name="TCustomer")
public class Customer {
	@Column(name="customer_id")
	private String customerId;

	@Column(name="sales_id")
	private Sales sales;

	@Column(name="customer_name")
	private String customerName;

	@Column(name="customer_type")
	private String customerType;
 
	@Column(name="payment_type")
	private String paymentType;
 
	@Column(name="customer_contact")
	private String customerContact;

	@Column(name="customer_phone")
	private String customerPhone;

	@Column(name="customer_fax")
	private String customerFax;

	@Column(name="customer_email")
	private String customerEmail;

	@Column(name="customer_address")
	private String customerAddress;

	@Column(name="customer_create_time")
	private java.util.Date createdDate;

	@Column(name="customer_last_modified_time")
	private java.util.Date lastModifiedDate;

	public String getCustomerId() {
		return customerId;
	}

	public void setCustomerId(String customerId) {
		this.customerId = customerId;
	}

	public Sales getSales() {
		return sales;
	}

	public void setSales(Sales sales) {
		this.sales = sales;
	}

	public String getCustomerName() {
		return customerName;
	}

	public void setCustomerName(String customerName) {
		this.customerName = customerName;
	}

	public String getCustomerType() {
		return customerType;
	}

	public void setCustomerType(String customerType) {
		this.customerType = customerType;
	}

	public String getPaymentType() {
		return paymentType;
	}

	public void setPaymentType(String paymentType) {
		this.paymentType = paymentType;
	}

	public String getCustomerContact() {
		return customerContact;
	}

	public void setCustomerContact(String customerContact) {
		this.customerContact = customerContact;
	}

	public String getCustomerPhone() {
		return customerPhone;
	}

	public void setCustomerPhone(String customerPhone) {
		this.customerPhone = customerPhone;
	}

	public String getCustomerFax() {
		return customerFax;
	}

	public void setCustomerFax(String customerFax) {
		this.customerFax = customerFax;
	}

	public String getCustomerEmail() {
		return customerEmail;
	}

	public void setCustomerEmail(String customerEmail) {
		this.customerEmail = customerEmail;
	}

	public String getCustomerAddress() {
		return customerAddress;
	}

	public void setCustomerAddress(String customerAddress) {
		this.customerAddress = customerAddress;
	}

	public java.util.Date getCreatedDate() {
		if (createdDate == null) createdDate = new java.util.Date();
		return createdDate;
	}

	public void setCreatedDate(java.util.Date createdDate) {
		this.createdDate = createdDate;
	}

	public java.util.Date getLastModifiedDate() {
		if (lastModifiedDate == null) lastModifiedDate = new java.util.Date();
		return lastModifiedDate;
	}

	public void setLastModifiedDate(java.util.Date lastModifiedDate) {
		this.lastModifiedDate = lastModifiedDate;
	}
}